packages/pharos/src/components/toggle-button-group/pharos-toggle-button.ts

```diff
@@ -0,0 +1,158 @@
+import { html, LitElement, property } from 'lit-element';
+import type { TemplateResult, CSSResultArray } from 'lit-element';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { nothing } from 'lit-html';
+import { toggleButtonStyles } from './pharos-toggle-button.css';
+import { designTokens } from '../../styles/variables.css';
+import { customElement } from '../../utils/decorators';
+import FocusMixin from '../../utils/mixins/focus';
+
+import '../icon/pharos-icon';
+import type { IconName } from '../icon/pharos-icon';
+
+export type { IconName };
+
+/**
+ * Pharos toggle button component.
+ *
+ * @element pharos-toggle-button
+ *
+ * @slot - Contains the content of the button (the default slot).
+ *
+ */
+@customElement('pharos-toggle-button')
+export class PharosToggleButton extends FocusMixin(LitElement) {
```

Comment:
I see this component borrows a lot from the `pharos-button` (as it should). Instead of duplicating a lot of the code/styles you can instead `extend` the `PharosButton` class to add additional logic/styles on top of it. Check out the `PharosSidenavButton` class to see how this is done.

Reply:
done